pkg/overlay/overlay.go

```diff
@@ -46,24 +49,81 @@ func MountTemp(store storage.Store, containerId, source, dest string, rootUID, r
 		return mount, "", errors.Wrapf(err, "failed to create the overlay %s directory", workDir)
 	}
 
+	overlayOptions := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s,private", source, upperDir, workDir)
+
+	if unshare.IsRootless() {
+		mountProgram := ""
+
+		for _, i := range store.GraphOptions() {
+			s := strings.SplitN(i, "=", 2)
+			if len(s) != 2 {
+				continue
+			}
+			k := s[0]
+			v := s[1]
+			switch k {
```

Comment:
How about you setup a map
mountMap := map[string]bool {
               ".mount_program": true,
               "overlay.mount_program": true,
               "overlay2.mount_program":true, 
}
...
if mountMap[k] {
           mountProgram=v
           break
}

Reply:
amended in the last version